ts/sagas/startup.ts

```diff
@@ -136,16 +136,14 @@ function* initializeApplicationSaga(): IterableIterator<Effect> {
   // User is autenticated, session token is valid
   //
 
-  if (maybeSessionInformation.isSome()) {
-    // the wallet token is available,
-    // proceed with starting the "watch wallet" saga
-    const pagoPaClient: PagoPaClient = PagoPaClient(pagoPaApiUrlPrefix);
-    yield fork(
-      watchWalletSaga,
-      pagoPaClient,
-      maybeSessionInformation.value.walletToken
-    );
-  }
+  // the wallet token is available,
+  // proceed with starting the "watch wallet" saga
+  const pagoPaClient: PagoPaClient = PagoPaClient(pagoPaApiUrlPrefix);
+  yield fork(
+    watchWalletSaga,
+    pagoPaClient,
+    maybeSessionInformation.value.walletToken
```

Comment:
my point here is: since whenever `maybeSessionInformation.value.walletToken` changes we have to create a new `PagoPaClient`, why not encapsulate the token in the client instance?

```ts
const pagoPaClient: PagoPaClient = PagoPaClient(pagoPaApiUrlPrefix, maybeSessionInformation.value.walletToken);
```

does the `watchWalletSaga` needs to be aware that there is a `walletToken` ?

Reply:
The saga does need to be aware of `walletToken` for the reasons stated [here](https://github.com/teamdigitale/italia-app/pull/426#discussion_r213585043). As I said, it can be made unaware of the token, but at the cost of hiding part of the process (the token refresh mechanism) from the saga and integrating it in the client instead.